### PR TITLE
[String] Fix #54611 pluralization of -on ending words + singularization of -a ending foreign words

### LINKED ISSUE #54611: [String][EnglishInflector] Pluralize on words ending with -on mostly wrong
### Symfony version(s) affected

5.4 +

### Description

When using the EnglishInflector of the String component to pluralize words ending with -on the results are mainly wrong because of the follwing rule in the SINGULAR_MAP:
```php
        // bacteria (bacterium), criteria (criterion), phenomena (phenomenon)
        ['no', 2, true, true, 'a'],
```

This -on rule is not valid in most of the cases. The rule works for foreign (mainly greek) words. This is an irregular case not a standard.

We can already see 3 other rules for words ending in -pon, -son and -con. 

### How to reproduce

```php
// First, run "composer require symfony/string"
// Then, execute this file:
<?php
require_once __DIR__.'/vendor/autoload.php';
use Symfony\Component\String\Inflector\EnglishInflector;

// returns "butta" instead of "buttons"
echo (new EnglishInflector())->pluralize('button');
```

### Possible Solution

I'm not an english native-speaker nor an expert of english grammar but I think that it would be better to have a rule dedicated to the foreign word ending in -on, phenomenon and criterion (and maybe others) + a generic rule for words ending in -on.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by DesLynx**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Issues        | Fix #54611
| License       | MIT

* fix the pluralization of -on ending words as -ons (with exception for "criterion" and "phenomenon")

* fix singularization of -a ending foreign words (see https://english-zone.com/spelling/plurals.html)
image

note: I left "data" uninflected because it seems to me that changing it to the datum/data inflection could cause side effects as "data" is a widely used word (especially in the coding world).

* update the test suites (String component + deprecated Inflector component) according to the changes

### CODE REVIEW

**Comment by DesLynx** [comment]
The common typo issue reported by fabbot.io is not a typo here.

**Comment by DesLynx** [comment]
The `Unit Tests / Unit Tests (8.2, low-deps) (pull_request)` uses the String component v5.3 making the Inflector component test failed. Should I have targetted the 5.3 branch instead of the 5.4?

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
(you need to bump the version of the string component in the composer.json file to fix the deps=low tests)

**Comment by DesLynx** [comment]
@nicolas-grekas you lost me on this because I don't know to which version should I bump the string component in the composer file of the inflector component as the PR fixes the string component. I obviously can't use 5.4.x-dev ...

**Comment by nicolas-grekas** [comment]
The next version is 5.4.39

File: src/Symfony/Component/Inflector/composer.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
         "php": ">=7.2.5",
         "symfony/deprecation-contracts": "^2.1|^3",
         "symfony/polyfill-php80": "^1.16",
-        "symfony/string": "^5.3.10|^6.0"
+        "symfony/string": "^5.4.39|^6.0"
```
Comment (xabbuh) [inline src/Symfony/Component/Inflector/composer.json]:
```suggestion
        "symfony/string": "^5.4.39|^6.4.7"
```

**Review by xabbuh — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @DesLynx.

**Pull request merged**